### PR TITLE
check for lld in shell scripts

### PR DESCRIPTION
**Pull request by makslevental**

### CODE REVIEW

File: .github/workflows/buildAndTest.yml [hunk 1]
```diff
@@ -175,7 +175,7 @@ jobs:
             -DCMAKE_MODULE_PATH=`pwd`/../cmake/modulesXilinx \
             -DMLIR_DIR=../llvm/install/lib/cmake/mlir \
             -DLLVM_DIR=../llvm/install/lib/cmake/llvm \
-            -DCMAKE_LINKER=lld \
```
Comment (makslevental) [inline .github/workflows/buildAndTest.yml:178]: This is not the right thing https://stackoverflow.com/a/25274328

File: .github/workflows/buildAndTest.yml [hunk 2]
```diff
@@ -175,7 +175,7 @@ jobs:
...
+            -DLLVM_USE_LINKER=lld \
```
Comment (makslevental) [inline .github/workflows/buildAndTest.yml:178]: [this flag sets](https://github.com/llvm/llvm-project/blob/4aae5387a874a55ee491f5dc23ce0506c5cdc678/llvm/cmake/modules/HandleLLVMOptions.cmake#L361).

```
-DCMAKE_EXE_LINKER_FLAGS_INIT="-fuse-ld=lld"
-DCMAKE_MODULE_LINKER_FLAGS_INIT="-fuse-ld=lld"
-DCMAKE_SHARED_LINKER_FLAGS_INIT="-fuse-ld=lld"
```

File: utils/build-llvm-cross-local.sh [hunk 3]
```diff
@@ -57,18 +58,16 @@ cmake ../llvm \
   -DLLVM_BUILD_EXAMPLES=OFF \
   -DLLVM_BUILD_UTILS=ON \
   -DLLVM_INSTALL_UTILS=ON \
-  -DLLVM_USE_LINKER=lld \
   -DMLIR_ENABLE_BINDINGS_PYTHON=ON \
   -DCMAKE_INSTALL_PREFIX=$INSTALL_DIR \
-  -DLLVM_ENABLE_PROJECTS="clang;lld;mlir" \
```
Comment (makslevental) [inline utils/build-llvm-cross-local.sh:63]: Don't need these.

File: utils/build-llvm-cross-local.sh [hunk 4]
```diff
@@ -78,19 +77,19 @@ cmake ../llvm \
   -DMLIR_INCLUDE_INTEGRATION_TESTS=OFF \
   -DLINKER_SUPPORTS_COLOR_DIAGNOSTICS=OFF \
   -DLLVM_ENABLE_TERMINFO=OFF \
-  -DLLVM_DEFAULT_TARGET_TRIPLE="aarch64-linux-gnu" \
-  -DLLVM_TARGET_ARCH="AArch64" \
-  -DCMAKE_CXX_STANDARD=17 \
```
Comment (makslevental) [inline utils/build-llvm-cross-local.sh:83]: This is already specified in [`llvm/CMakeLists.txt`](https://github.com/llvm/llvm-project/blob/3fe69bab1149e20f87ac31f8b2557baa88fcdd14/llvm/CMakeLists.txt#L70).

**Review by keryell — APPROVED** [review]
Thanks or the cleaning.

**Pull request merged**